esp/esdllib/esdl_script.cpp

```diff
@@ -0,0 +1,919 @@
+/*##############################################################################
+
+    HPCC SYSTEMS software Copyright (C) 2020 HPCC Systems®.
+
+    Licensed under the Apache License, Version 2.0 (the "License");
+    you may not use this file except in compliance with the License.
+    You may obtain a copy of the License at
+
+       http://www.apache.org/licenses/LICENSE-2.0
+
+    Unless required by applicable law or agreed to in writing, software
+    distributed under the License is distributed on an "AS IS" BASIS,
+    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+    See the License for the specific language governing permissions and
+    limitations under the License.
+############################################################################## */
+
+#include "espcontext.hpp"
+#include "esdl_script.hpp"
+#include "wsexcept.hpp"
+
+interface IEsdlTransformOperation : public IInterface
+{
+    virtual const char *queryMergedTarget() = 0;
+    virtual bool process(IEspContext * context, IPropertyTree *content, IXpathContext * xpathContext) = 0;
+    virtual void toDBGLog() = 0;
+};
+
+IEsdlTransformOperation *createEsdlTransformOperation(IPropertyTree *element, const StringBuffer &prefix);
+
+inline void esdlOperationError(int code, const char *op, const char *msg, const char *traceName, bool exception)
+{
+    StringBuffer s("ESDL Script: ");
+    if (!isEmptyString(traceName))
+        s.append(" '").append(traceName).append("' ");
+    if (!isEmptyString(op))
+        s.append(" ").append(op).append(" ");
+    s.append(msg);
+    if(exception)
+        throw MakeStringException(code, "%s", s.str());
+
+    IERRLOG("%s", s.str());
+}
+
+inline void esdlOperationError(int code, const char *op, const char *msg, bool exception)
+{
+    esdlOperationError(code, op, msg, "", exception);
+}
+
+static inline const char *checkSkipOpPrefix(const char *op, const StringBuffer &prefix)
+{
+    if (prefix.length())
+    {
+        if (!hasPrefix(op, prefix, true))
+        {
+            DBGLOG(1,"Unrecognized script operation: %s", op);
+            return nullptr;
+        }
+        return (op + prefix.length());
+    }
+    return op;
+}
+
+static inline StringBuffer &makeOperationTagName(StringBuffer &s, const StringBuffer &prefix, const char *op)
+{
+    return s.append(prefix).append(op);
+}
+
+class CEsdlTransformOperationBase : public CInterfaceOf<IEsdlTransformOperation>
+{
+protected:
+    StringAttr mergedTarget;
+    StringAttr tagname;
```

Comment:
Use m_* style for members to match other classes?

Reply:
Ok.  Not a huge fan of that style but I mostly tried to be consistent with what was there.  May have missed a few.